qemu/lib/iozone.py

```diff
@@ -0,0 +1,133 @@
+                                                        timeout=self.timeout)
+        self.results_path = os.path.join(self.test.resultsdir, 'raw_output_%s'
+                                         % self.test.iteration)
+        logging.info("Write results to %s" % self.results_path)
+        #omit write results error
+        try:
+            with open(self.results_path, 'w') as file:
+                file.write(output)
+        except:
+            pass
+
+        return status, output
+
+    def after_run_iozone(self):
+        """
+        Clean up iozone env/post result after iozone test
+        """
+        raise NotImplementedError()
+
+
+class Iozone_Win(IozoneTest):
+    """
+    Post result after iozone test.
+    """
+    def __init__(self, vm, params, test):
+        super(Iozone_Win, self).__init__(vm, params, test)
+
+    def after_run_iozone(self):
+        """
+        Post result after iozone test, generate a series graphs
+        """
+        # use this function, params "-i" for iozone is not suggested.
+        # suggest usage:iozone.exe -azR -r 64k -n 125M -g 512M -M -b
+        # I:\iozone.xls -f I:\testfile".
+        # Otherwise, your result can't be analysed correctly.
+        if self.params.get("post_results", "no") == "yes":
+            logging.info("Generate graph of test result")
+            analysisdir = os.path.join(self.test.resultsdir, 'analysis_%s'
+                                       % self.test.iteration)
+            iozone_analyzer = postprocess_iozone.IOzoneAnalyzer(list_files=[self.results_path],
+                                                                output_dir=analysisdir)
+            iozone_analyzer.analyze()
+            iozone_plotter = postprocess_iozone.IOzonePlotter(results_file=self.results_path,
+                                                              output_dir=analysisdir)
+            iozone_plotter.plot_all()
+
+
+class Iozone_Linux(IozoneTest):
+    """
+    Install iozone test tool before test, clean up iozone packets after test
+    """
+    def __init__(self, vm, params, test):
+        super(Iozone_Linux, self).__init__(vm, params, test)
+        stress_type = "iozone"
+        self.iozone_work_path = params.get("iozone_work_path", "/home/iozone3_458/src/current")
+        download_url = params.get("download_url_iozone",
+                                  "http://www.iozone.org/src/stable/iozone3_458.tar")
+        make_cmds = params.get("make_cmds_iozone", "make linux")
+        stress_cmds = params.get("stress_cmds_iozone", "./iozone")
+        stress_args = params.get("stress_args_iozone", "")
+        work_path = params.get("work_path_iozone", "src/current")
+        self.stress_iozone_linux = utils_test.VMStress(vm, stress_type, download_url, make_cmds, stress_cmds,
+                                                       stress_args, work_path)
+
+    def before_run_iozone(self):
+        """
+        Install iozone tool before test
+        """
```

Comment:
Hi aliang, if can have a function to automatic allocation mountpoint at before_run_iozone()?
like for windows guest, it should be a prepare steps before iozone run, if we need to run iozone on multidisks, we'll no need to define multi mountpoint in cfg file manually~

Reply:
@peixiu, if can have a function to automatic allocation mountpoint?
  A: format_widnows_disk() in https://github.com/avocado-framework/avocado-vt/pull/1556 has realized it, with a mountpint_list returned.
  So, we just need to call it in run_iozone_in_guest().